pkg/swap/block.go

```diff
@@ -18,12 +36,12 @@ func (g *GethClient) FetchBlockTime(ctx context.Context) (int64, error) {
 		return 0, fmt.Errorf("fetch latest block timestamp: %w", err)
 	}
 
-	timestampPrevious, err := g.fetchBlockTimestamp(ctx, latestBlockNumber-1)
+	timestampPrevious, err := g.fetchBlockTimestamp(ctx, latestBlockNumber-o.offset)
```

Comment:
what if latestBlockNumber is less than o.offset ? this could occur in custom environment with new deployments of geth.

Reply:
also by default o.offset is zero, the option to define it is not used.

Reply:
> what if latestBlockNumber is less than o.offset ? this could occur in custom environment with new deployments of geth.

Well, I was hoping that RPC will return error in this case. But maybe I can implement some retry logic that can half the offset until it finds the value? What do you think?

Reply:
i think using zero should be ok. 
you can do something like:
```
if latestBlockNumber < o.offset {
  g.fetchBlockTimestamp(ctx, 0)
}
```

Reply:
For some reason, when block 0 is used to query the Geth node, it returns the timestamp 1582023590, which corresponds to February 18, 2020. To avoid this issue, I limited the maximum offset to half of the latest block number.